internal/broker/instance_create.go

```diff
@@ -338,6 +343,33 @@ func (b *ProvisionEndpoint) validateAndExtract(details domain.ProvisionDetails,
 		}
 	}
 
+	if IsFreemiumPlan(details.PlanID) && b.config.OnlyOneFreePerGA {
+		count, err := b.instanceArchivedStorage.TotalNumberOfInstancesArchivedForGlobalAccountID(ersContext.GlobalAccountID, FreemiumPlanID)
+		if err != nil {
+			return ersContext, parameters, fmt.Errorf("while checking if a free Kyma instance existed for given global account: %w", err)
+		}
+
+		if count > 0 {
+			logger.Info("Provisioning Free SKR rejected, such instance was already created for this Global Account")
+			return ersContext, parameters, fmt.Errorf("free Kyma was created for the global account, but there is only one allowed")
+		}
+
+		filter := dbmodel.InstanceFilter{
+			GlobalAccountIDs: []string{ersContext.GlobalAccountID},
+			PlanIDs:          []string{FreemiumPlanID},
+			States:           []dbmodel.InstanceState{dbmodel.InstanceSucceeded},
+		}
+		_, _, count, err = b.instanceStorage.List(filter)
```

Comment:
currently, the instances archived is empty (there is no data). After a deprovisioning, the instance is deleted - this ios the reason we have to check operations instead of instances. Check, if there is an operation "provisioning" which has seuceeded for such global maccount